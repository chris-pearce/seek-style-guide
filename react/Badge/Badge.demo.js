import React from 'react';
import PropTypes from 'prop-types';
import { Card, Text, PageBlock, Section, Badge } from 'seek-style-guide/react';

export default {
  route: '/badge',
  title: 'Badge',
  component: Badge,

  initialProps: {
    children: '2 new',
    info: true
  },
  options: [
    {
      label: 'Text',
      type: 'radio',
      states: [
        ,
        {
          label: 'Info',
          transformProps: props => ({
            ...props,
            info: true
          })
        },
        {
          label: 'Accent',
          transformProps: props => ({
            ...props,
            accent: true
          })
        },
        {
          label: 'Secondary',
          transformProps: props => ({
            ...props,
            secondary: true
          })
        },
        {
          label: 'Positive',
          transformProps: props => ({
            ...props,
            positive: true
          })
        },
        {
          label: 'Critical',
          transformProps: props => ({
            ...props,
            critical: true
          })
        }
      ]
    },
    {
      label: 'States',
      type: 'checkbox',
      states: [
        {
          label: 'Strong',
          transformProps: props => ({
            ...props,
            strong: true
          })
        }
      ]
    }
  ]
};
