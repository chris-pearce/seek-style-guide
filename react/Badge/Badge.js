// @flow

import React, { PureComponent } from 'react';
import classnames from 'classnames';

import styles from './Badge.less';

type Props = {
  children: React$Node,
  strong?: boolean,
  info?: boolean,
  accent?: boolean,
  critical?: boolean,
  secondary?: boolean,
  positive?: boolean
};

export default class Badge extends PureComponent<Props> {
  static displayName = 'Badge';

  props: Props;

  render() {
    const {
      info,
      accent,
      critical,
      secondary,
      positive,
      strong,
      children
    } = this.props;

    return (
      <div
        className={classnames({
          [styles.root]: true,
          [styles.info]: info,
          [styles.accent]: accent,
          [styles.critical]: critical,
          [styles.secondary]: secondary,
          [styles.positive]: positive,
          [styles.strong]: strong
        })}
        {...this.props}
      >
        {children}
      </div>
    );
  }
}
