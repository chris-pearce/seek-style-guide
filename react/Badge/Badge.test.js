import React from 'react';
import { shallow } from 'enzyme';
import Badge from './Badge';

const renderBadge = props => shallow(<Badge {...props}>2 new</Badge>);

test('should render Badge', () => {
  const badge = renderBadge();
  expect(badge).toMatchSnapshot();
});
