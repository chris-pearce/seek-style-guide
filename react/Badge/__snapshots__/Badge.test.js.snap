// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Badge 1`] = `
<div
  className="Badge__root"
>
  2 new
</div>
`;
